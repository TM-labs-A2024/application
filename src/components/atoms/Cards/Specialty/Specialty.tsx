import { Link } from '@chakra-ui/next-js'
import { Text, Avatar } from '@chakra-ui/react'
import { Patient } from '@src/types'
import NextLink from 'next/link'
import React from 'react'

import Chevron from '../../../../../public/static/icons/chevron.svg'

export default function Specialty({
  name,
  id,
  patient
}: {
  name: string
  id: string
  patient?: Patient
}) {
  return (
    <Link
      as={NextLink}
      href={patient ? `/especialidad/${patient?.id}/${id}` : `/especialidad/${id}`}
      className="border-1 flex w-full rounded-md border border-gray-400 p-4"
      data-testid="specialty-card"
    >
      <div className="flex w-full flex-row items-center justify-between">
        <div className="flex flex-row items-center">
          <Avatar name={name} src="/" />
          <Text className="ml-4">{name}</Text>
        </div>
        <Chevron />
      </div>
    </Link>
  )
}
