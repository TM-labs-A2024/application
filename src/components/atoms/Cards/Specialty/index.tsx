import Specialty from './Specialty'

export default Specialty
