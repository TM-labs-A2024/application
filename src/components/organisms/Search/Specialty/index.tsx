import SpecialtySearch from './SpecialtySearch'

export default SpecialtySearch
