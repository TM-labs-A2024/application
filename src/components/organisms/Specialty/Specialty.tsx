import { ArrowBackIcon } from '@chakra-ui/icons'
import {
  IconButton,
  Text,
  Avatar,
  Tabs,
  TabList,
  TabPanels,
  Tab,
  TabPanel,
  TabIndicator,
  Button,
  Icon
} from '@chakra-ui/react'
import SearchInputComponent from '@components/atoms/SearchInput'
import EvolutionsList from '@components/molecules/EvolutionsList'
import { CardEvolution } from '@src/types'
import { isAndroid, isIOS, isMobile } from '@utils/index'
import { useRouter } from 'next/navigation'
import { useRouter as queryRouter } from 'next/router'
import React, { useState, useEffect, useMemo } from 'react'
import { RiHealthBookFill } from 'react-icons/ri'

export default function Specialty({
  context
}: {
  context: {
    isPatient: boolean
    isDoctor: boolean
    specialty: {
      id: string
      name: string
    }
    data: {
      evolutions: CardEvolution[]
      orders: CardEvolution[]
      tests: CardEvolution[]
    }
    currentTab: number
  }
}) {
  // --- Hooks -----------------------------------------------------------------
  const router = useRouter()
  const route = queryRouter()
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  const { isPatient, isDoctor, specialty, data, currentTab } = context
  const [_window, setWindow] = useState({ screen: { availWidth: 999 } })
  const [tabIndex, setTabIndex] = useState(currentTab)
  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------
  useEffect(() => {
    setWindow(window)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])
  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  const patientId = useMemo(() => !isPatient && route?.query.slug?.[0], [isPatient, route])
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div
      className={`flex h-screen w-screen flex-col p-8 lg:px-96 ${isIOS() ? 'pt-20' : 'pt-8'}`}
      data-testid="specialty"
    >
      <div className="mb-8 flex flex-row items-center justify-between">
        <div className="flex flex-row items-center justify-start gap-4">
          <IconButton
            size="xl"
            aria-label="back"
            variant="link"
            icon={<ArrowBackIcon />}
            onClick={() => {
              patientId
                ? router.push(`/especialidades/${patientId}`)
                : router.push('/especialidades')
            }}
          />
          <Text className="font-medium">{specialty.name}</Text>
        </div>
        <Avatar name={specialty.name} src="/" size="sm" />
      </div>
      {data.evolutions.length > 0 && (
        <SearchInputComponent
          placeholder="Buscar"
          className="mb-8 w-full"
          onClick={() => {
            patientId
              ? router.push(`/especialidad/busqueda/${patientId}/${specialty.id}`)
              : router.push(`/especialidad/busqueda/${specialty.id}`)
          }}
        />
      )}
      <Tabs
        isFitted={!isMobile(_window)}
        variant="unstyled"
        className="h-[80%]"
        defaultIndex={currentTab}
        onChange={(index) => setTabIndex(index)}
      >
        <TabList>
          {(isPatient || isDoctor) && (
            <Tab fontSize={isMobile(_window) ? '0.8rem' : '1rem'} width="25%">
              Evoluciones
            </Tab>
          )}
          <Tab fontSize={isMobile(_window) ? '0.8rem' : '1rem'} width="50%">
            Ordenes médicas
          </Tab>
          {(isPatient || isDoctor) && (
            <Tab fontSize={isMobile(_window) ? '0.8rem' : '1rem'} width="25%">
              Análisis
            </Tab>
          )}
        </TabList>
        <TabIndicator mt="-1.5px" height="2px" bg="black" borderRadius="1px" />

        <TabPanels className="h-full pb-0">
          {(isPatient || isDoctor) && (
            <TabPanel className={`${isAndroid() ? 'h-[75%]' : 'h-[90%]'} overflow-scroll`}>
              <EvolutionsList evolutions={data.evolutions} />
            </TabPanel>
          )}
          <TabPanel className={`${isAndroid() ? 'h-[75%]' : 'h-[90%]'} overflow-scroll`}>
            {data.orders.length > 0 && <EvolutionsList evolutions={data.orders} />}
            {data.orders.length === 0 && (
              <div
                className="flex h-full w-full items-center justify-center"
                data-testid="orders-empty-state"
              >
                <Icon fontSize="xxx-large" as={RiHealthBookFill} />
                <Text textAlign="center" mt={4}>
                  No hay órdenes asociadas aún.
                </Text>
                {isDoctor && (
                  <Button
                    mt={4}
                    onClick={() =>
                      router.push(`/crear-adjunto/${patientId}/${specialty.id}?type=order`)
                    }
                  >
                    Nueva orden
                  </Button>
                )}
              </div>
            )}
          </TabPanel>
          {(isPatient || isDoctor) && (
            <TabPanel className={`${isAndroid() ? 'h-[75%]' : 'h-[90%]'} overflow-scroll`}>
              {data.tests.length > 0 && <EvolutionsList evolutions={data.tests} />}
              {data.tests.length === 0 && (
                <div
                  className="flex h-full w-full items-center justify-center"
                  data-testid="tests-empty-state"
                >
                  <Icon fontSize="xxx-large" as={RiHealthBookFill} />
                  <Text textAlign="center" mt={4}>
                    No hay análisis asociados aún.
                  </Text>
                  <Button
                    mt={4}
                    onClick={() =>
                      router.push(`/crear-adjunto/${patientId}/${specialty.id}?type=order`)
                    }
                  >
                    Nuevo análisis
                  </Button>
                </div>
              )}
            </TabPanel>
          )}
          {isDoctor && tabIndex === 0 && (
            <Button
              className="mt-4 w-full"
              onClick={() => router.push(`/crear-historia/${patientId}`)}
            >
              Nueva evolución
            </Button>
          )}
          {isDoctor && tabIndex === 1 && (
            <Button
              className="mt-4 w-full"
              onClick={() => router.push(`/crear-adjunto/${patientId}/${specialty.id}?type=order`)}
            >
              Nueva orden
            </Button>
          )}
          {isDoctor && tabIndex === 2 && (
            <Button
              className="mt-4 w-full"
              onClick={() => router.push(`/crear-adjunto/${patientId}/${specialty.id}?type=test`)}
            >
              Nuevo análisis
            </Button>
          )}
        </TabPanels>
      </Tabs>
    </div>
  )
}
