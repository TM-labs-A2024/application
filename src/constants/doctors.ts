export const doctors = [
  {
    id: '1000000',
    institutionId: '1000000',
    credentials: '1000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '2000000',
    institutionId: '2000000',
    credentials: '2000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '3000000',
    institutionId: '3000000',
    credentials: '3000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '4000000',
    institutionId: '4000000',
    credentials: '4000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '5000000',
    institutionId: '5000000',
    credentials: '5000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '6000000',
    institutionId: '6000000',
    credentials: '6000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '7000000',
    institutionId: '7000000',
    credentials: '7000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '8000000',
    institutionId: '8000000',
    credentials: '8000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '9000000',
    institutionId: '9000000',
    credentials: '9000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '10000000',
    institutionId: '10000000',
    credentials: '10000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '11000000',
    institutionId: '11000000',
    credentials: '11000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '12000000',
    institutionId: '12000000',
    credentials: '12000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '13000000',
    institutionId: '13000000',
    credentials: '13000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '14000000',
    institutionId: '14000000',
    credentials: '14000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '15000000',
    institutionId: '15000000',
    credentials: '15000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '16000000',
    institutionId: '16000000',
    credentials: '16000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '17000000',
    institutionId: '17000000',
    credentials: '17000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '18000000',
    institutionId: '18000000',
    credentials: '18000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '19000000',
    institutionId: '19000000',
    credentials: '19000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '20000000',
    institutionId: '20000000',
    credentials: '20000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '21000000',
    institutionId: '21000000',
    credentials: '21000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '22000000',
    institutionId: '22000000',
    credentials: '22000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522',
    pending: true,
    patientPending: true
  },
  {
    id: '23000000',
    institutionId: '23000000',
    credentials: '23000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '24000000',
    institutionId: '24000000',
    credentials: '24000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '25000000',
    institutionId: '25000000',
    credentials: '25000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '26000000',
    institutionId: '26000000',
    credentials: '26000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '27000000',
    institutionId: '27000000',
    credentials: '27000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '28000000',
    institutionId: '28000000',
    credentials: '28000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '29000000',
    institutionId: '29000000',
    credentials: '29000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '30000000',
    institutionId: '30000000',
    credentials: '30000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '31000000',
    institutionId: '31000000',
    credentials: '31000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '32000000',
    institutionId: '32000000',
    credentials: '32000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '33000000',
    institutionId: '33000000',
    credentials: '33000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '34000000',
    institutionId: '34000000',
    credentials: '34000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '35000000',
    institutionId: '35000000',
    credentials: '35000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '36000000',
    institutionId: '36000000',
    credentials: '36000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '37000000',
    institutionId: '37000000',
    credentials: '37000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '38000000',
    institutionId: '38000000',
    credentials: '38000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '39000000',
    institutionId: '39000000',
    credentials: '39000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '40000000',
    institutionId: '40000000',
    credentials: '40000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '41000000',
    institutionId: '41000000',
    credentials: '41000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '42000000',
    institutionId: '42000000',
    credentials: '42000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '43000000',
    institutionId: '43000000',
    credentials: '43000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '44000000',
    institutionId: '44000000',
    credentials: '44000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '45000000',
    institutionId: '45000000',
    credentials: '45000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '46000000',
    institutionId: '46000000',
    credentials: '46000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '47000000',
    institutionId: '47000000',
    credentials: '47000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  },
  {
    id: '48000000',
    institutionId: '48000000',
    credentials: '48000000',
    specialties: ['1', '4'],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com',
    phoneNumber: '4127775522'
  }
]
