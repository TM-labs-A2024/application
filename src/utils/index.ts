import { sendEmail } from './email'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export { isWeb, isIOS, isAndroid, isMobile, sendEmail }
