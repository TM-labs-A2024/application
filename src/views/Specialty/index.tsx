import Specialty from '@src/components/organisms/Specialty'
import { CardEvolution } from '@src/types'
import React from 'react'

export default function SpecialtyView({
  context
}: {
  context: {
    isPatient: boolean
    isDoctor: boolean
    specialty: {
      id: string
      name: string
    }
    data: {
      evolutions?: CardEvolution[]
      orders?: CardEvolution[]
      tests?: CardEvolution[]
    }
    currentTab: number
  }
}) {
  return <Specialty context={context} />
}
