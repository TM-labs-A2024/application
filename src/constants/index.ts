import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
