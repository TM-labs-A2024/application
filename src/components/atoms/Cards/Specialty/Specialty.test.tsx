import { patient } from '@src/constants'
import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import Specialty from './Specialty'

const specialty = {
  name: 'Alergología',
  id: '10000000'
}

const specialtyWithPatient = {
  name: 'Alergología',
  id: '10000000',
  patient
}

describe('Atoms > Cards > Specialty test', () => {
  test('The card shows the name of the specialty', async () => {
    render(<Specialty {...specialty} />)

    expect(await screen.findByText(specialty.name)).toBeInTheDocument()
  })

  test('The url matches the expected in the pacients flow', async () => {
    render(<Specialty {...specialty} />)

    const enlace = await screen.findByRole('link')
    expect(enlace).toHaveAttribute('href', `/especialidad/${specialty.id}`)
  })

  test('The url matches the expected in the doctors flow', async () => {
    render(<Specialty {...specialtyWithPatient} />)

    const enlace = await screen.findByRole('link')
    expect(enlace).toHaveAttribute(
      'href',
      `/especialidad/${specialtyWithPatient?.patient?.govId}/${specialtyWithPatient.id}`
    )
  })

  test('Matches the snapshot', async () => {
    render(<Specialty {...specialty} />)

    expect(await screen.findByTestId('specialty-card')).toMatchSnapshot()
  })
})
