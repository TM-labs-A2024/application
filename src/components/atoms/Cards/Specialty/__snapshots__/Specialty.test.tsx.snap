// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > Cards > Specialty test Matches the snapshot 1`] = `
<a
  class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
  data-testid="specialty-card"
  href="/especialidad/10000000"
>
  <div
    class="flex w-full flex-row items-center justify-between"
  >
    <div
      class="flex flex-row items-center"
    >
      <span
        class="chakra-avatar css-1mw972v"
      >
        <div
          aria-label="Alergología"
          class="chakra-avatar__initials css-1ebyn6"
          role="img"
        >
          A
        </div>
      </span>
      <p
        class="chakra-text ml-4 css-0"
      >
        Alergología
      </p>
    </div>
    <div />
  </div>
</a>
`
